manager/integration/tests/test_recurring_job.py

```diff
@@ -178,9 +180,10 @@ def test_recurring_job(set_random_backupstore, client, volume_name):  # NOQA
     for snapshot in snapshots:
         if snapshot.removed is False:
             count += 1
-    # 2 from job_snap, 1 from job_backup, 2 from job_backup2, 1 volume-head
-    assert count == 6
+    # 2 from job_snap, 1 from job_backup, 1 from job_backup2, 1 volume-head
+    assert count == 5
 
+    time.sleep(20)
```

Comment:
Why we need to have sleep here? Does the case fail before due to no sleep?

I would suggest having a retry (like 3 times and sleep in less seconds) mechanism instead of just long sleep. (Please check if we already have retry function able to reuse)

Reply:
Changed the logic here.  The problem arises when running NFS, backup seems sometimes slow and cause count to be one less than expected.